src/infra/flow-system-inmem/tests/tests/test_flow_service_inmem.rs

```diff
@@ -414,6 +426,293 @@ async fn test_manual_trigger() {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+#[test_log::test(tokio::test)]
+async fn test_manual_trigger_compaction() {
+    let harness = FlowHarness::new();
+
+    let foo_id = harness.create_root_dataset("foo").await;
+    let bar_id = harness.create_root_dataset("bar").await;
+
+    // Note: only "foo" has auto-schedule, "bar" hasn't
+    harness
+        .set_dataset_flow_schedule(
+            harness.now_datetime(),
+            foo_id.clone(),
+            DatasetFlowType::HardCompaction,
+            Duration::try_milliseconds(90).unwrap().into(),
+        )
+        .await;
+    harness.eager_dependencies_graph_init().await;
+
+    let foo_flow_key: FlowKey =
+        FlowKeyDataset::new(foo_id.clone(), DatasetFlowType::HardCompaction).into();
+    let bar_flow_key: FlowKey =
+        FlowKeyDataset::new(bar_id.clone(), DatasetFlowType::HardCompaction).into();
+
+    let test_flow_listener = harness.catalog.get_one::<FlowSystemTestListener>().unwrap();
+    test_flow_listener.define_dataset_display_name(foo_id.clone(), "foo".to_string());
+    test_flow_listener.define_dataset_display_name(bar_id.clone(), "bar".to_string());
+
+    // Remember start time
+    let start_time = harness
+        .now_datetime()
+        .duration_round(Duration::try_milliseconds(SCHEDULING_ALIGNMENT_MS).unwrap())
+        .unwrap();
+
+    // Run scheduler concurrently with manual triggers script
+    tokio::select! {
+        // Run API service
+        res = harness.flow_service.run(start_time) => res.int_err(),
+
+        // Run simulation script and task drivers
+        _ = async {
+                // Task 0: "foo" start running at 10ms, finish at 20ms
+                let task0_driver = harness.task_driver(TaskDriverArgs {
+                    task_id: TaskID::new(0),
+                    dataset_id: Some(foo_id.clone()),
+                    run_since_start: Duration::try_milliseconds(10).unwrap(),
+                    finish_in_with: Some((Duration::try_milliseconds(10).unwrap(), TaskOutcome::Success(TaskResult::Empty))),
+                    max_slice_size: None,
+                    max_slice_records: None,
+                });
```

Comment:
No tasks would auto-start for compaction flow - you are not auto-scheduling anything for this type.